packages/worker/src/worker.ts

```diff
@@ -1492,6 +1492,29 @@ export class Worker {
     return takeUntil(this.stateSubject.pipe(filter((value) => value === state)));
   }
 
+  /**
+   * {@link run | Run} the Worker until `fnOrPromise` completes.
+   *
+   * @returns the result of `fnOrPromise`
+   *
+   * Throws on fatal Worker errors.
+   */
+  async runUntil<R>(fnOrPromise: Promise<R> | (() => Promise<R>)): Promise<R> {
```

Comment:
Isn't this just `this.run().finally(typeof fnOrPromise === 'function' ? fnOrPromise : () => fnOrPromise)`?